libtextsecure/message_receiver.js

```diff
@@ -144,6 +144,24 @@ MessageReceiver.prototype.extend({
         //   processing is complete by the time it runs.
         Promise.all(incoming).then(queueDispatch, queueDispatch);
     },
+    drain: function() {
+        var incoming = this.incoming;
+        this.incoming = [];
+
+        var queueDispatch = function() {
+            // resetting count to zero so everything queued after this starts over again
+            this.count = 0;
+
+            return this.addToQueue(function() {
+              console.log('drained');
```

Comment:
I think it's because the comment was copied (along with the rest of the method) from `onEmpty()` above. The key to this method is its return value, which is a promise that resolves when the shutdown is complete.